ksql-cli/src/main/java/io/confluent/ksql/cli/console/CommentStripper.java

```diff
@@ -98,7 +98,7 @@ private String strip() {
     private String trimComment(final int partStart) {
       final String part = line.substring(partStart, pos - 1).trim();
 
-      final int newLine = line.indexOf(System.lineSeparator(), pos + 1);
+      final int newLine = line.indexOf("\n", pos + 1);
```

Comment:
Taken at face value, the old code looked more cross-platform independent. Would you mind explaining the workings behind this change?

Reply:
Yeah, this was the most PITA part of the whole fix, and the only place i think where i changed the code-under-test rather than the test code itself.  IIRC, after debugging deeper than anyone should ever have to into the nested and repeated calls of the various parser classes, it's the case that this code only ever gets passed the output of an invocation of a JLine parser, which has already normalized the embedded line feeds to \n. 
Note that there's confusion in the code about the meaning of variables called "line" - sometimes it's "a string of characters delimited by a newline", as you might intuit, and sometimes "a logical line, maybe containing embedded newlines which we sometimes want to retain and sometimes don't, delimited by the line termination character (the ; we use to indicate end-of-sql-statement)".